### PR TITLE
[dvdnav] use highlight area palette for active overlay colors

### PR DESCRIPTION
**Pull request by enen92**
## Description

DVD menu overlays got broken for windows on https://github.com/xbmc/xbmc/pull/21154 which attempted to remove one of our dvdnav patches. For some reason the pci struct doesn't get filled with the properties we were trying to access. Oddly enough, it works fine on linux...

Diving a bit more on this, instead of abusing struct members that likely shouldn't be publicly accessed I've looked into how VLC does it. It's much more clean than our approach since the color information for the active button is already part of the `highlight_area` returned by `dvdnav_get_highlight_area`:

https://github.com/videolan/vlc/blob/79d852d5be832729375860f9d9fce0e3d391aa89/modules/access/dvdnav.c#L1341-L1379

Allow us to finally drop that ugly method too.

Run time tested on windows and linux.

**Current master on Windows:**
image

**PR on Windows**
image

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamNavigator.cpp [hunk 1]
```diff
@@ -1159,8 +1159,7 @@ bool CDVDInputStreamNavigator::SetAngle(int angle)
 
 bool CDVDInputStreamNavigator::GetCurrentButtonInfo(CDVDOverlaySpu* pOverlayPicture, CDVDDemuxSPU* pSPU, int iButtonType)
 {
-  int alpha[2][4];
-  int color[2][4];
+  int colorAndAlpha[4][4];
```
Comment (howie-f) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamNavigator.cpp:1162]: maybe use a c++ `std::array<int, 4 * 4> colorAndAlpha;` and then access its elements via `colorAndAlpha.at(x * 4 + y) = ...` as this provides bounds checking. well not sure if it's worth for such a small thing, but...

**Review by howie-f — COMMENTED** [review]
i can't comment on the fix for windows, just a suggestion for the array.

File: xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamNavigator.cpp (hunk 1 above)
Comment (enen92) [reply xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamNavigator.cpp:1162]: Compared to the ridiculous amount of time I've spent on this I'll gladly try to implement it :)

**Review by howie-f — APPROVED** [review]
ok, it seems that the issue just doesn't occur with the only iso i have left here to test :( thanks a lot for fixing. this is fine for me to go in, forget about the array ;)

**Pull request merged**